dsgrid/registry/dimension_mapping_registry_manager.py

```diff
@@ -144,17 +140,103 @@ def _check_records_against_dimension_records(self, config):
             if diff:
                 dim_id = from_dimension.model.dimension_id
                 raise DSGInvalidDimensionMapping(
-                    f"Dimension mapping has invalid 'to' records: dimension_id={dim_id} {diff}"
+                    f"Dimension mapping={mapping.filename} has invalid 'to_id' records: {diff}, "
+                    f"they are missing from dimension_id={dim_id}"
                 )
 
-    def validate_records(self, config: DimensionMappingsConfig, warn_only=False):
+    def validate_records(self, config: DimensionMappingsConfig):
         """Validate dimension mapping records.
 
         Check:
-        - from_id and to_id column names
-        - check for duplicate IDs and log a warning if they exist (sometimes we want them to exist if it is an aggregation)
+        - duplicate records in from_id and to_id columns per mapping archetype
+        - sum of from_fraction by from_id per mapping archetype
+        - special check for mapping_type=duplication
+
         """
-        pass
+        for mapping in config.model.mappings:
+            if "one_to_one" in mapping.archetype.value:
```

Comment:
This relies on these enum values producing multiple matches, right? Seems a bit non-straightforward. Python enums are very flexible. You could probably add a method that returns allow_dup_from/to_records.